### ISSUE TITLE
object-curly-newline / array-bracket-newline: Allow single elements with newline

### ISSUE THREAD

**Issue by ThiefMaster** [eslint/eslint#10093]
Currently there doesn't seem to be an option that covers all these ways of writing object/array literals:

```javascript

const a = {};
const b = {a: 1, b: 2};
const c = {c: 3};
const d = {
    d: 4,
    e: 5
};
const e = {
    e: 1
};
```

I could use `consistent` style but I don't want to allow any multi-element no-newline literals. Did I miss the option to do what I want or is this currently not possible? If it's the latter, I think it would be a useful improvement to these rules.

**Comment by kadamwhite** [comment]
```
		"object-curly-newline": [ "error", {
			"minProperties": 2,
			"consistent": true
		} ]
```
is probably the closest to what you want. `consistent` will suppress warnings on single-line objects with _fewer_ than `minProperties`; so in this case the only issue it would flag in your code would I believe be `const b = { a: 1, b: 2 };`.

**Comment by kadamwhite** [comment]
(I definitely had trouble with this too, I think the docs could stand to be improved to make this clearer but I'm not sure I have a good wording suggestion myself)

**Comment by kaicataldo** [comment]
Apologies that we lost track of this. @ThiefMaster has this been resolved?

**Comment by ThiefMaster** [comment]
For me the "consistent" option seems fine for both arrays and objects. At least I didn't encounter any problems with it since enabling it.

**Comment by kaicataldo** [comment]
Great! Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
